### PR TITLE
Develop jh

### PR DESCRIPTION
**Pull request by johannah-pik**
New features:

- Improved ECEMF scenarios
- Include option to adjust efficiency based on mitigation ambition either by supply of parameters for a logistic trend or an annual improvement rate  
- Bugfix knitting function for CompareScenarios Transport 
- Apply mean to remove technology column from loadFactor also for shipping to REMIND (before unique was used that led to duplicates) 
- Bugfix reportEDGET setNames from magclass was not imported and set to Null in the beginning 
- Add emissions and new cost plot to CompareScenarios

### CODE REVIEW

File: R/finalSteps.R [hunk 1]
```diff
@@ -146,10 +146,11 @@ toolCreateOutput <- function(logit_params, pref_data, ptab4W, vot_data, NEC_data
   ## remove column not needed in REMIND from int_dat
   int_dat[, MJ_km := NULL]
 
-  ## remove redundant columns from load factor
-  loadFactor = unique(loadFactor[year >= 1990, c("region", "year", "loadFactor", "vehicle_type")])
+  ## remove technology column from load factor
+  load_Factor <- load_Factor[year >= 1990, .(loadFactor = mean(loadFactor)),
```
Comment (Loisel) [inline R/finalSteps.R:150]: Also quite useful. I thought I tackled that one already :thinking:

File: R/generateEDGEdata.R [hunk 2]
```diff
@@ -29,13 +30,14 @@
...
 
 
-toolGenerateEDGEdata <- function(input_folder, output_folder, cache_folder = NULL,
-                                 SSP_scen = "SSP2", tech_scen = "Mix", smartlifestyle = FALSE,
-                                 storeRDS = FALSE,
-                                 gdxPath = NULL,
-                                 preftab = NULL, plot.report = FALSE,
-                                 mitab4W.path = NULL, mitab.path = NULL,
-                                 ssp_demreg.path = NULL, regional_demreg.path = NULL, FEPricetab = NULL) {
+generateEDGEdata <- function(input_folder, output_folder, cache_folder = NULL,
```
Comment (Loisel) [inline R/generateEDGEdata.R:33]: is there a good reason to rename this function? the tool prefix helps to have it monitored by `mrremind`...

File: R/incotrend.R [hunk 3]
```diff
@@ -276,6 +276,7 @@ BEV,Electric
 NG,Liquids
 Hybrid Electric,Liquids")
     mimap <- fread(system.file("extdata", "mitigation-techmap.csv", package="edgeTransport"))
+    mimap <- mimap[!FV_vehvar == "LDV|4W"]
```
Comment (Loisel) [inline R/incotrend.R:279]: seems legit. Did this lead to issues? SWs are zero for those no?

File: inst/extdata/edget-mitigation.csv [hunk 4]
```diff
@@ -1,1985 +1,1985 @@
...
-SSP2,Nav_HighH2_LifestCha,rich,S3,Cycle,,,,1,2050,10
-SSP2,Nav_HighH2_LifestCha,rich,S3,Domestic Aviation,,,,0.25,2035,10
-SSP2,Nav_HighH2_LifestCha,rich,S3,Domestic Ship,,,,1,2050,10
-SSP2,Nav_HighH2_LifestCha,rich,S3,Freight Rail,,,,1,2050,10
-SSP2,Nav_HighH2_LifestCha,rich,S3,HSR,,,,1,2050,10
-SSP2,Nav_HighH2_LifestCha,rich,S3,Passenger Rail,,,,1,2050,10
-SSP2,Nav_HighH2_LifestCha,rich,S3,Walk,,,,1,2050,10
-SSP2,Nav_HighH2_LifestCha,rich,S3,trn_freight_road,,,,0.5,2035,10
-SSP2,Nav_HighH2_LifestCha,rich,S3,trn_pass_road,,,,0.5,2035,10
+SSP_scenario,tech_scenario,regioncat,level,subsector_L3,subsector_L2,FV_vehvar,FV_techvar,target,symmyr,speed,,,,,,
```
Comment (Loisel) [inline inst/extdata/edget-mitigation.csv:1]: please remove the empty columns from the table to have a useful changeset

File: inst/extdata/inconv_factor.csv [hunk 5]
```diff
@@ -1,569 +1,569 @@
...
-SSP2,ECEMF_HighH2_HighEff,ratioPHEV,1.5
-SSP2,ECEMF_HighH2_LifestCha,startYeBEV,2025
-SSP2,ECEMF_HighH2_LifestCha,startValBEV,1
-SSP2,ECEMF_HighH2_LifestCha,targetYeBEV,2050
-SSP2,ECEMF_HighH2_LifestCha,targetValBEV,1.5
-SSP2,ECEMF_HighH2_LifestCha,startYeICE,2020
-SSP2,ECEMF_HighH2_LifestCha,targetYeICE,2027
-SSP2,ECEMF_HighH2_LifestCha,targetValICE,0.5
-SSP2,ECEMF_HighH2_LifestCha,ratioPHEV,1.5
+SSPscen,techscen,param,value,,,
```
Comment (Loisel) [inline inst/extdata/inconv_factor.csv:1]: same here

**Review by Loisel — CHANGES_REQUESTED** [review]
Tables seem to be appended with empty columns, please remove.

File: R/adjust_intensity.R [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                      by = c("region", "vehicle_type", "technology")]
+        #Keep the intensity from 2050 onward constant
+        intensity_new_after2050 <- rbindlist(lapply(unique(intensity_new[year > 2050]$year), function(x){intensity_new[year == 2050][, year := x]}))
+        intensity_new <- rbind(intensity_new[year <= 2050], intensity_new_after2050)
+        intensity_new[, c("FV_vehvar", "FV_techvar", "level", "target", "symmyr", "speed", "SSP_scenario", "tech_scenario", "annual_rate") := NULL]
+      }
+
+      #Merge with unaffected intensity data
+      intensity <- merge(intensity, intensity_new, by = c("region", "technology", "vehicle_type", "year", "subsector_L1", "subsector_L2", "subsector_L3", "sector", "sector_fuel"), all = TRUE)
+      intensity[, EJ_Mpkm_final := ifelse(is.na(EJ_Mpkm_final.y), EJ_Mpkm_final.x, EJ_Mpkm_final.y)][, c("EJ_Mpkm_final.x", "EJ_Mpkm_final.y") := NULL]
```
[the remark this answers is not in the document]
Comment (johannah-pik) [reply R/adjust_intensity.R:40]: Hmm I dont really get it..

File: R/generateEDGEdata.R (hunk 2 above)
Comment (johannah-pik) [reply R/generateEDGEdata.R:33]: Well spotted, that was not my intention

File: R/incotrend.R (hunk 3 above)
Comment (johannah-pik) [reply R/incotrend.R:279]: No it was more a safety measure as I introduced these entries in the techmap and didn't have the time to track all data wrangling steps in incotrend (I am not so familiar with the notation you are using)

File: R/generateEDGEdata.R [hunk 7]
```diff
@@ -262,6 +264,15 @@ toolGenerateEDGEdata <- function(input_folder, output_folder, cache_folder = NUL
   if(storeRDS)
     saveRDS(prefs, file = level1path("prefs.RDS"))
 
+  #Optional Energy Intensity improvements after 2020 depending on the tech Scen
+  if(is.null(int_improvetab)){
```
[the remark this answers is not in the document]
Comment (Loisel) [reply R/generateEDGEdata.R:268]: maybe just close the bracket of the if-clause after line 271.

[the remark this answers is not in the document]
Comment (Loisel) [reply R/generateEDGEdata.R:268]: maybe also test again since the `adjust_intensity` function has most likely not been executed in your last test.

**Comment by Loisel** [comment]
One general comment: Pleas do not lump to many different *things* into one PR. From this I would make at least 3 PRs, one for the intensity interface, one for the new scenarios and one for the rest.

File: R/generateEDGEdata.R (hunk 7 above)
[the remark this answers is not in the document]
Comment (johannah-pik) [reply R/generateEDGEdata.R:268]: Thats not true, its executed as long as you dont supply a special intensity tab but just load the default one from the package. It is tested line by line. I agree that in the long run it is better to decide for either supplying annual rates or parameters for a logistic trend but for the moment I have to keep the old version

**Comment by johannah-pik** [comment]
Agree. But we do the exact same thing when doing a PR to the master don't we?

**Comment by Loisel** [comment]
the PR to master is what is called a "release". It always contains a range of features and fixes and goes with a new version number. PRs to develop are ideally feature specific such that one can revert or cherry-pick features for releases.

**Review by Loisel — APPROVED** [review]

**Comment by johannah-pik** [comment]
Okay :)

**Pull request merged**